### PR TITLE
RORDEV 493 fail if upload (or git tag) fails

### PR DESCRIPTION
**Pull request by sscarduzio**

### CODE REVIEW

File: ci/ci-lib.sh [hunk 1]
```diff
@@ -2,28 +2,52 @@
...
+    fi
+    echo "Deleting tag $GIT_TAG"
+    if [[ "$(uname -s)" == *"Linux"* ]]; then
+        git tag -d $GIT_TAG && git push origin :refs/tags/$GIT_TAG || (echo "Failed to delete tag $GIT_TAG" && exit 1)
+        return 0
     fi
     return 0
 }
 
+file . || (echo "FATAL: file command does not work or is not installed!" && exit 1)
```
Comment (coutoPL) [inline ci/ci-lib.sh:50]: do we need it?

File: ci/ci-lib.sh [hunk 2]
```diff
@@ -2,28 +2,52 @@
...
-        git config --global user.name "Travis CI"
-        git tag $GIT_TAG -a -m "Generated tag from TravisCI build $TRAVIS_BUILD_NUMBER"
-        git push origin $GIT_TAG
+        git tag $GIT_TAG -a -m "Generated tag from TravisCI build $TRAVIS_BUILD_NUMBER" &&
+        git push origin $GIT_TAG || exit 1
+    fi
+    return 1
+}
+
+function __tag_exists {
```
Comment (coutoPL) [inline ci/ci-lib.sh:27]: do we need __ prefix?

File: ci/ci-lib.sh [hunk 3]
```diff
@@ -2,28 +2,52 @@
...
     echo "Tagging as $GIT_TAG"
     if [[ "$(uname -s)" == *"Linux"* ]]; then
-        git remote set-url origin git@github.com:sscarduzio/elasticsearch-readonlyrest-plugin.git
-        git config --global push.default matching
-        git config --global user.email "builds@travis-ci.com"
-        git config --global user.name "Travis CI"
-        git tag $GIT_TAG -a -m "Generated tag from TravisCI build $TRAVIS_BUILD_NUMBER"
-        git push origin $GIT_TAG
+        git tag $GIT_TAG -a -m "Generated tag from TravisCI build $TRAVIS_BUILD_NUMBER" &&
+        git push origin $GIT_TAG || exit 1
```
Comment (coutoPL) [inline ci/ci-lib.sh:22]: no error log on fail?

**Review by coutoPL — CHANGES_REQUESTED** [review]

File: ci/ci-lib.sh (hunk 2 above)
Comment (sscarduzio) [reply ci/ci-lib.sh:27]: it's a way to say "private" in a language where there's no private methods, reminiscent from Python.

File: ci/ci-lib.sh (hunk 1 above)
Comment (sscarduzio) [reply ci/ci-lib.sh]: This is checking a dependency for ci-uploader.sh, should belong to that file, indeed. 👍

**Comment by coutoPL** [comment]
does it mean this is not ready yet?

**Comment by sscarduzio** [comment]
What I did so far is functional to have this working (making the git command available), yet, something else is on the way: at the moment, Github does not get the right credentials for tagging

```
Host key verification failed.
fatal: Could not read from remote repository.
```

**Comment by coutoPL** [comment]
can we mark the PR as a draft?

**Comment by sscarduzio** [comment]
Yeah this thing didn't work. Will need to drastically simplify this whole dumpster fire and start over

**Pull request closed without merging**